acl.go

```diff
@@ -0,0 +1,104 @@
+package imap
+
+import (
+	"strings"
+)
+
+// IMAP4 ACL extension (RFC 2086)
+
+type RightSet string
+
+type Right byte
+
+const (
+	// Standard rights
+	RightLookup     = Right('l') // mailbox is visible to LIST/LSUB commands
+	RightRead       = Right('r') // SELECT the mailbox, perform CHECK, FETCH, PARTIAL, SEARCH, COPY from mailbox
+	RightSeen       = Right('s') // keep seen/unseen information across sessions (STORE SEEN flag)
+	RightWrite      = Right('w') // STORE flags other than SEEN and DELETED
+	RightInsert     = Right('i') // perform APPEND, COPY into mailbox
+	RightPost       = Right('p') // send mail to submission address for mailbox, not enforced by IMAP4 itself
+	RightCreate     = Right('c') // CREATE new sub-mailboxes in any implementation-defined hierarchy
+	RightDelete     = Right('d') // STORE DELETED flag, perform EXPUNGE
+	RightAdminister = Right('a') // perform SETACL
+
+	AllStandardRights = RightSet("lrswipcda")
+)
+
+type RightsIdentifier string
+
+const RightsIdentifierAnyone = RightsIdentifier("anyone")
+
+type RightModification byte
+
+const (
+	RightModificationReplace = RightModification(0)
+	RightModificationAdd     = RightModification('+')
+	RightModificationRemove  = RightModification('-')
+)
+
+// NewRights converts rights string into RightModification and RightSet with validation
+func NewRights(rights string) (RightModification, RightSet) {
```

Comment:
This doesn't need to be part of our public API I think (could be moved to our `internal` package).